### DISCUSSION TITLE
sometimes 503 even if the service is healthy

### DISCUSSION THREAD

**Discussion by AhmedKamal20** [nginx-proxy/nginx-proxy#2544] [Q&A]
Hi,
I have a Django container that I use the reverse proxy with.
Normally Djnago reload the server when the code changes, right after the server is ready, I can hit the host and it works normally.

But if I hit the host during this reloading, it gives me 503 which is expected of course, but after the server is up, I have to wait like 30 seconds to be able to hit it again.

So If I waited the reload, I can call the host immediately , but If by mistake I called it during being down, I have to wait much loner.

Is there some kind of caching or something that cause this ?

**Comment by buchdag** [comment]
That's strange, there is nothing that should cache the upstream response, we're not using `proxy_cache` at all 🤔

Would you be able to confirm that your Django application actually respond when queried directly while the proxy still give a 503 ?

**Reply by AhmedKamal20** [reply]
Yes, confirmed 
Once it shows this :
```
Django version 5.0.6, using settings 'config.settings'
Starting development server at http://0.0.0.0:5000/
Quit the server with CONTROL-C.
```
It's ready

Also, as I mentioned in the original post, it will work immediately if I wait until the above message appears.
The problem starts to appear when I'm rushing it, and start calling the server before this message, then the message appears and the server is ready but it gives me 503 for like 30 seconds before it will work again

**Reply by AhmedKamal20** [reply]
Sorry it gives me `502 Bad Gateway` not 503
